standalone/src/models/product-selection/types.ts

```diff
@@ -3,24 +3,36 @@ import type {
   ProductSelectionDraft,
 } from '@commercetools/platform-sdk';
 import type { TBuilder } from '@/core';
-import { TLocalizedStringGraphql } from '@/models/commons';
+import {
+  TCtpProductSelection,
+  TCtpCreateProductSelectionDraft,
+  TCtpProductOfSelection,
+} from '@/graphql-types';
 
-//ProductSelectionDraft
+/**
+ * @deprecated use `TProductSelectionDraftRest` or `TProductSelectionDraftGraphql` instead
+ */
 export type TProductSelectionDraft = ProductSelectionDraft;
-export type TProductSelectionDraftBuilder = TBuilder<TProductSelectionDraft>;
-export type TCreateProductSelectionDraftBuilder =
-  () => TProductSelectionDraftBuilder;
-export type TProductSelectionDraftGraphql = Omit<
-  TProductSelectionDraft,
-  'name'
-> & {
-  name: TLocalizedStringGraphql;
-};
-
-//ProductSelection
+/**
+ * @deprecated use `TProductSelectionRest` or `TProductSelectionGraphql` instead
+ */
 export type TProductSelection = ProductSelection;
-export type TProductSelectionBuilder = TBuilder<TProductSelection>;
-export type TCreateProductSelectionBuilder = () => TProductSelectionBuilder;
-export type TProductSelectionGraphql = TProductSelection & {
-  __typename: 'ProductSelection';
-};
+
+// REST types
+export type TProductSelectionDraftRest = ProductSelectionDraft;
+export type TProductSelectionRest = ProductSelection;
+
+// GraphQL types
+export type TProductSelectionDraftGraphql = TCtpCreateProductSelectionDraft;
+export type TProductSelectionGraphql = TCtpProductSelection;
+export type TProductOfSelectionGraphql = TCtpProductOfSelection;
```

Comment:
The `ProductOfSelection` model should have its own `types.ts` file.

Reply:
I see. But why it's not the case for the draft model then?

Reply:
The draft model is a sibling of the "main" model. They relate to the very same entity in the APIs.
A submodel is a different model that is related to its parent.